tests/lib/newlib/newlib_hook/src/main.c

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2021 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <sys/unistd.h>
+#include <sys/signal.h>
+#include <sys/stat.h>
+
+/**
+ * @brief Verify _open() stub in newlib libc-hooks.c
+ *
+ * @details The compiler warning when call open(),
+ * use fopen() instead.
+ */
+void test_newlib_stub_open(void)
+{
+	FILE *fp = NULL;
+
+	fp = fopen("../prj.conf", "r");
+	zassert_is_null(fp, "open file error");
+}
+
+/**
+ * @brief Verify _close() stub in newlib libc-hooks.c
+ *
+ * @details The default weak function return -1.
+ */
+void test_newlib_stub_close(void)
+{
+	int ret = 0;
+
+	ret = close(-1);
+	zassert_equal(ret, -1, "closed a invalid fd");
+}
+
+/**
+ * @brief Verify _getpid() stub in newlib libc-hooks.c
+ *
+ * @details The default weak function return 0.
+ */
+void test_newlib_stub_getpid(void)
+{
+	pid_t pid;
+
+	pid = getpid();
+	zassert_false(pid, "getpid failed");
+}
+
+/**
+ * @brief Verify _stat() stub in newlib libc-hooks.c
+ *
+ * @details The default weak function return 0.
+ */
+void test_newlib_stub_fstat(void)
+{
+	struct stat status;
+	int ret;
+
+	ret = fstat(-1, &status);
+	zassert_false(ret, "fstat failed");
```

Comment:
`ret` is not a boolean type.
```suggestion
	zassert_equal(ret, 0, "fstat failed");
```